python_modules/libraries/dagster-components/dagster_components/core/dsl_schema.py

```diff
@@ -24,36 +25,63 @@ def to_automation_condition(self) -> AutomationCondition:
         return getattr(AutomationCondition, self.type)(**self.params)
 
 
-class AssetSpecProcessorModel(BaseModel):
+class AssetSpecProcessor(ABC, BaseModel):
     target: str = "*"
-    strategy: Union[Literal["replace"], Literal["merge"]] = "replace"
-    description: Optional[str] = None
-    metadata: Optional[Mapping[str, Any]] = None
-    group_name: Optional[str] = None
-    tags: Optional[Mapping[str, str]] = None
-    automation_condition: Optional[AutomationConditionModel] = None
 
-    def _props(self) -> Mapping[str, Any]:
-        return {
-            **self.model_dump(exclude={"target", "strategy"}),
-            "automation_condition": self.automation_condition.to_automation_condition()
-            if self.automation_condition
-            else None,
-        }
+    def _attributes(self) -> Mapping[str, Any]:
+        return {**self.model_dump(exclude={"target", "operation"}, exclude_unset=True)}
 
-    @suppress_dagster_warnings
-    def _apply_to_spec(self, spec: AssetSpec) -> AssetSpec:
-        if self.strategy == "replace":
-            return spec.replace_attributes(**self._props())
-        else:
-            return spec.merge_attributes(**self._props())
+    @abstractmethod
+    def _apply_to_spec(self, spec: AssetSpec) -> AssetSpec: ...
+
+    def apply(self, defs: Definitions) -> Definitions:
+        target_selection = AssetSelection.from_string(self.target, include_sources=True)
+        target_keys = target_selection.resolve(defs.get_asset_graph())
 
-    def transform(self, defs: Definitions) -> Definitions:
-        """Applies the specified transformation to the asset specs in the given definitions."""
         mappable = [d for d in defs.assets or [] if isinstance(d, (AssetsDefinition, AssetSpec))]
-        mapped_assets = map_asset_specs(self._apply_to_spec, mappable)
+        mapped_assets = map_asset_specs(
+            lambda spec: self._apply_to_spec(spec) if spec.key in target_keys else spec, mappable
+        )
+
         assets = [
             *mapped_assets,
             *[d for d in defs.assets or [] if not isinstance(d, (AssetsDefinition, AssetSpec))],
         ]
         return replace(defs, assets=assets)
+
+
+class ReplaceAttributes(AssetSpecProcessor):
+    # default operation is replace
+    operation: Literal["replace"] = "replace"
```

Comment:
There is a pretty good argument I think to make the "default" to be "set" and have tags and metadata have "merge" semantics and call "merge" "override"